assets/js/hooks/useScreenplayContext.tsx

```diff
@@ -0,0 +1,226 @@
+import * as React from "react";
+import { createGenericContext } from "../utils/createGenericContext";
+import { Place } from "../models/place";
+import { Alert } from "../models/alert";
+import { ScreensByAlert } from "../models/screensByAlert";
+import { useReducer } from "react";
+import {
+  MODES_AND_LINES,
+  SCREEN_TYPES,
+  STATUSES,
+} from "../constants/constants";
+import { BannerAlert } from "../components/Dashboard/AlertBanner";
+
+interface Props {
+  children: React.ReactNode;
+}
+
+type DirectionID = 0 | 1;
+
+interface FilterValue {
+  label: string;
+  ids: string[];
+  color?: string;
+}
+
+interface PageProps {
+  sortDirection: DirectionID;
+  modeLineFilterValue: FilterValue;
+  screenTypeFilterValue: FilterValue;
+  statusFilterValue: FilterValue;
+}
+
+interface StateContextType {
+  places: Place[];
+  alerts: Alert[];
+  screensByAlertMap: ScreensByAlert;
+  placesPageDirectionId: DirectionID;
+  placesPage: PageProps & {
+    showScreenlessPlaces: boolean;
+    activeEventKeys: string[];
+  };
+  alertsPage: PageProps;
+  bannerAlert?: BannerAlert;
+}
+
+const reducer = (state: any, action: any) => {
+  switch (action.type) {
```

Comment:
Hmm I could see an alternative approach where we have reducers for each page. Mainly I think it would just remove the need for the if-else blocks and reduce the size of the reducer(s) but it might be a little cleaner looking. Not sure if that would work in practice without playing around with `context` a bit, but what I have in mind is have like a `dashboardReducer` with the SET_PLACES, SET_ALERTS, and SET_SCREENS_BY_ALERT actions and then have a `placesReducer` and `alertsReducer` with basically the same set of actions but export those to their respective components and remove the need to pass the action.page field.

Or are we only allowed to have one reducer for the entire context?

Reply:
I gave that a shot. It felt messy putting it together, but I think clarity may be a little better. It's all in one commit so if we don't like it, it's easy to change back. What do you think?